src/tjoperators.jl

```diff
@@ -0,0 +1,639 @@
+#= Operators that act on t-J-type models
+i.e. the local hilbert space consists of 
+=#
+module TJOperators
+
+using TensorKit
+
+export tj_space
+export c_num, u_num, d_num, c_num_hole
+export S_x, S_y, S_z, S_plus, S_min
+export u_plus_u_min, d_plus_d_min
+export u_min_u_plus, d_min_d_plus
+export u_min_d_min, d_min_u_min
+export c_plus_c_min, c_min_c_plus, c_singlet
+export S_plusmin, S_minplus, S_exchange
+
+export nꜛ, nꜜ, nʰ, n
+export Sˣ, Sʸ, Sᶻ, S⁺, S⁻
+export u⁺u⁻, d⁺d⁻, u⁻u⁺, d⁻d⁺, u⁻d⁻, d⁻u⁻
+export c⁺c⁻, c⁻c⁺
+export S⁻⁺, S⁺⁻
+
+"""
+    tj_space(particle_symmetry::Type{<:Sector}, spin_symmetry::Type{<:Sector}; slave_fermion::Bool = false)
+
+Return the local hilbert space for a t-J-type model with the given particle and spin symmetries.
+The possible symmetries are 
+- Particle number: `Trivial`, `U1Irrep`;
+- Spin: `Trivial`, `U1Irrep`, `SU2Irrep`.
+
+Setting `slave_fermion = true` switches to the slave-fermion basis. 
+
+- basis states for `slave_fermion = false`: 
+    |∅⟩, |↑⟩, |↓⟩
+- basis states for `slave_fermion = true`: (c_σ = h† b_σ; holon h is fermionic, spinon b_σ is bosonic): 
+    |h⟩ = h†|∅⟩, |↑'⟩ = (b↑)†|∅⟩, |↓'⟩ = (b↓)†|∅⟩
```

Comment:
Similarly, I want to change
```suggestion
    |0⟩ = h†|vac⟩, |↑⟩ = (b↑)†|vac⟩, |↓⟩ = (b↓)†|vac⟩
```
so that in both cases the basis states are denoted as |0⟩, |↑⟩ and |↓⟩.